src/shared/components/results/TaxonomyLevelsSuggestion.tsx

```diff
@@ -6,29 +8,68 @@ import {
 } from './SearchSuggestions';
 import OrganismSuggestion from './OrganismSuggestion';
 
-const TaxonomyLevelsSuggestion = ({ query }: { query: string }) => {
+import useDataApi from '../../hooks/useDataApi';
+
+import apiUrls from '../../config/apiUrls';
+
+import { SearchResults } from '../../types/results';
+import { UniProtkbAPIModel } from '../../../uniprotkb/adapters/uniProtkbConverter';
+import { Namespace } from '../../types/namespaces';
+
+const TaxonomyLevelsSuggestion = ({
+  query,
+  total,
+}: {
+  query: string;
+  total: number;
+}) => {
+  const [showTaxonSuggestion, setShowTaxonSuggestion] = useState(false);
   const { modifiedQuery, searchValue } = modifyQueryWithSuggestions(
     query,
     'taxon',
     taxonHierarchySearchTerms
   );
 
+  const searchParams = new URLSearchParams({
+    query: `${modifiedQuery}`,
+  });
+
+  const { data } = useDataApi<SearchResults<UniProtkbAPIModel>>(
+    `${apiUrls.search(Namespace.uniprotkb)}?${searchParams}`
+  );
+
+  useEffect(() => {
+    if (data?.results.length && data?.results.length !== total) {
+      setShowTaxonSuggestion(true);
+    }
+  }, [data, total]);
```

Comment:
Any reason why this is in a `useEffect`?
This is derived from data and from a prop (`total`) so it could probably just be:
```ts
const showTaxonSuggestion = data?.results.length && data?.results.length !== total;
```